### PR TITLE
First draft of the "On how to implement a code of conduct" post

### PR DESCRIPTION
**Pull request by vanbrabantf**
This is the first draft of the "On how to implement a code of conduct" post

### CODE REVIEW

**Comment by vanbrabantf** [comment]
@andreascreten Could you please review if this is something we want to put on the company blog? Even though I've made it as neutral as possible it's still a controversial topic.

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+author: frederick
+excerpt: "Recently there has been a lot of commotion around code of conducts and if a user group/open-source project should implement them or not. This is not a post about that, we shall not be discussing the ethics or possible political impact this brings with it. We will however look into a starting point on **how** to implement them."
+---
+
+
+## Preface
+
+Recently there has been a lot of commotion around code of conducts and if a user group/open-source project should implement them or not. This is not a post about that, we shall not be discussing the ethics or possible political impact this brings with it. We will however look into a starting point on **how** to implement them.
+
+Please not that I am in no way an expert in this field, this blogpost is the result of endless talks with colleagues / friends and people I bugged on Slack (extra thanks to [Paul Hallett](https://twitter.com/phalt_) and [Jack Skinner](https://twitter.com/developerjack)).
```
Comment (maartenscholz) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: Please note

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  * Be: awesome.
+  * Don't be: not awesome.
+
+That's all good, but know that you'll be spending more of your time arguing about what passes as "Being awesome" then doing the thing you want to do. Holding conferences or writing code.
+
+So if you want to save yourself some trouble and or time down the line, a good idea is to implement something that's already thought out and proven to work.
+
+### 4. What's out there?
+
+Luckily there are some easy to adopt CoC's out there, for example [geekfeminism](http://geekfeminism.wikia.com/) has a conference blueprint that can be used to roll your own: [conferences CoC](http://geekfeminism.wikia.com/wiki/Conference_anti-harassment/Policy), if that one doesn't fit your bill there are others out there for example: [Conference Code of Conduct](http://confcodeofconduct.com/).
```
Comment (maartenscholz) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: easy `ways` to

**Comment by andreascreten** [comment]
Looks good to me. But @madewithlove/development please also read it. I am maybe not the best one to judge.

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+ * Read them.
+ * Take one that you think fits your project/conference best.
+ * Possibly adapt it to your needs. (if you adapt it, make it clear you've done so)
+
+Please note that I don't particularly want to say pick this one over that one, this is your choice here.
+
+### 6. Implement it
+
+Ok you've chosen the one you'll wonna go with. Now it's the moment you want to share it with the world. Made sure you clearly show that you have a CoC and that you expect people to follow it. They can't follow it if they don't know about it.
```
Comment (maartenscholz) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: `They can't follow it if they don't know about it.` feels like weird `it` sentence, maybe smth like `People can't follow a CoC if they know it exists`

Comment (miggynascimento) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: wanna or want to. :)

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title:  "On how to implement a code of conduct"
+date: 2016-01-29
+author: frederick
+excerpt: "Recently there has been a lot of commotion around code of conducts and if a user group/open-source project should implement them or not. This is not a post about that, we shall not be discussing the ethics or possible political impact this brings with it. We will however look into a starting point on **how** to implement them."
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: "look into" Don't know if look into is the right word here. Maybe: "We will however propose a starting point on ..."

not sure.

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+Please note that I don't particularly want to say pick this one over that one, this is your choice here.
+
+### 6. Implement it
+
+Ok you've chosen the one you'll wonna go with. Now it's the moment you want to share it with the world. Made sure you clearly show that you have a CoC and that you expect people to follow it. They can't follow it if they don't know about it.
+
+### 7. Communicate
+
+You've chosen a Coc, or not, you've put it out there, we are done right?
```
Comment (miggynascimento) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: CoC

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+So without further ado, I've made an easy 7 step program to follow along (this is the perfect time to fetch coffee or tea):
+
+## Steps
+
+### 1. What is a CoC?
+
+This is where the bulk of the drama comes from; most people are not really informed about what a Code of Conduct really is or start to overthink the concept. In essence it's actually really simple:
+
+A CoC or a code of conduct boils down to basically a list of rules, these rules can be as verbose or short as you like, as strict or as loose as you like. It's just an outline of how people should behave when they attend your conference or contribute to your open-source project.
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: either: `boils down to` or `basically`, not both

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: These rules ... (start of new sentence)

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+### 6. Implement it
+
+Ok you've chosen the one you'll wonna go with. Now it's the moment you want to share it with the world. Made sure you clearly show that you have a CoC and that you expect people to follow it. They can't follow it if they don't know about it.
+
+### 7. Communicate
+
+You've chosen a Coc, or not, you've put it out there, we are done right?
+
+Well no. Open-source is all about community. A project is never an island. There will be reactions on your decisions (if there aren't you should ask for them), and now it's the time to discuss it with the community. How you handle them is again up to you, but it could be wise to be open to them (if you think they're good) and be willing to adjust some parts.
```
Comment (miggynascimento) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: now is the time... flows a bit smoother

Comment (maartenscholz) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: `How you handle them is again up to you` =>
`Again, how you handle them is up to you`
reads better I think, could be wrong @miggynascimento might know

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+A CoC or a code of conduct boils down to basically a list of rules, these rules can be as verbose or short as you like, as strict or as loose as you like. It's just an outline of how people should behave when they attend your conference or contribute to your open-source project.
+
+Lots of places have a code of conduct, like for example a bar, think "we don't serve people without shoes".
+
+### 2. Do I need a CoC?
+
+You might react with: "Well we've never had any problems before and I don't see the point in having one". Sure, don't adopt one. The world isn't going to end if you don't adopt one.
+
+On the other hand a CoC can have the advantage that everyone clearly knows what's up. So if you turn into problems later down the line, you can easily redirect back to the rules.
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: `redirect back` don't know if this is being used in English. It's not HTTP :stuck_out_tongue: 

Maybe: `refer to the rules`

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+Ok you've chosen the one you'll wonna go with. Now it's the moment you want to share it with the world. Made sure you clearly show that you have a CoC and that you expect people to follow it. They can't follow it if they don't know about it.
+
+### 7. Communicate
+
+You've chosen a Coc, or not, you've put it out there, we are done right?
+
+Well no. Open-source is all about community. A project is never an island. There will be reactions on your decisions (if there aren't you should ask for them), and now it's the time to discuss it with the community. How you handle them is again up to you, but it could be wise to be open to them (if you think they're good) and be willing to adjust some parts.
+
+These things are like code, they can always be improved upon.
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: These `things`, the rules or entire CoCs?

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+##Addendum
+
+And that's it, a lot of drama about something simple right?
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: don't feed the trolls. I'd leave this sentence out

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+
+As you might have noticed, this is kind of a difficult topic and I am in no way an expert on this stuff. If you found some inaccuracies or want to discuss this further (keep it civil please) you can post in the comment section below or find me on [twitter](https://twitter.com/theedonian).
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: :+1: for `keep it civil, please`

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: Add that comma`,` though

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+On the other hand a CoC can have the advantage that everyone clearly knows what's up. So if you turn into problems later down the line, you can easily redirect back to the rules.
+
+Just make up your mind. As stated before, this post is not about whether or not you should implement one.
+
+If you don't want to implement one, you can skip to step 7.
+
+### 3. Should I write one myself?
+
+As always there is a lot of debate around this. You could if you want, but the general opinion is the same as on encryption, "Don't invent your own". Writing your own is pretty catchy (believe me I've tried), there are lots of things to think about and frankly you probably want a CoC that's clear about it's contents.
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: believe me`,` I've tried

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: There are lots (new sentence)

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: about `its` contents - classic mistake :wink:

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md (hunk 2 above)
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: CoC's or CoCs. I don't think the former is right.

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: others out there <comma or period> for example:

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+### 5. Pick one
+
+So now comes the difficult part, choose one. This means:
+
+ * Read them.
+ * Take one that you think fits your project/conference best.
+ * Possibly adapt it to your needs. (if you adapt it, make it clear you've done so)
+
+Please note that I don't particularly want to say pick this one over that one, this is your choice here.
```
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: I don't particularly want to say pick this one over that one
->
 I don't want to favor one over the other

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md (hunk 3 above)
Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: the one you'll wonna go with
a bit hard to read. Easier: `Ok, you've chosen your CoC. Now it's the moment...`

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: Now it's the moment
->
Now is the moment you want to...

Comment (hannesvdvreken) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: Made sure you clearly show
->
Ma`k`e sure you clearly show

**Comment by xavierbarbosa** [comment]
Nice way to approach this topic. :+1:

File: _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md (hunk 9 above)
Comment (vanbrabantf) [inline _posts/2016-01-29-on-how-to-implement-a-code-of-conduct.md]: Well this is a bit tricky, at point 7 you can either have a CoC or not. so I guess the concept of the CoC and it's rules or not

**Pull request merged**